homeassistant/components/tmb/sensor.py

```diff
@@ -0,0 +1,125 @@
+"""Support for TMB (Transports Metropolitans de Barcelona) Barcelona public transport."""
+from datetime import timedelta
+import logging
+
+from requests import HTTPError
+from tmb import IBus
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import ATTR_ATTRIBUTION, CONF_NAME
+from homeassistant.exceptions import Unauthorized
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTRIBUTION = "Data provided by Transport Metropolitans de Barcelona"
+
+ICON = "mdi:bus-clock"
+
+CONF_APP_ID = "app_id"
+CONF_APP_KEY = "app_key"
+CONF_LINE = "line"
+CONF_BUS_STOP = "stop"
+CONF_BUS_STOPS = "stops"
+ATTR_BUS_STOP = "stop"
+ATTR_LINE = "line"
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=60)
+
+LINE_STOP_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_BUS_STOP): cv.string,
+        vol.Required(CONF_LINE): cv.string,
+        vol.Optional(CONF_NAME): cv.string,
+    }
+)
+
+ROUTES_SCHEMA = vol.All(cv.ensure_list, [LINE_STOP_SCHEMA])
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_APP_ID): cv.string,
+        vol.Required(CONF_APP_KEY): cv.string,
+        vol.Optional(CONF_BUS_STOPS): ROUTES_SCHEMA,
+    }
+)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the sensors."""
+    ibus_client = IBus(config[CONF_APP_ID], config[CONF_APP_KEY])
+
+    sensors = []
+
+    for line_stop in config.get(CONF_BUS_STOPS):
+        line = line_stop[CONF_LINE]
+        stop = line_stop[CONF_BUS_STOP]
+        if line_stop.get(CONF_NAME):
+            name = f"{line} - {line_stop[CONF_NAME]} ({stop})"
+        else:
+            name = f"{line} - {stop}"
+        sensors.append(TMBSensor(ibus_client, stop, line, name))
+
+    if sensors:
```

Comment:
This guard isn't necessary; `add_entities` will automatically return if an empty list is passed.

Reply:
Fixed, didn't know that, thanks!